### DISCUSSION TITLE
Add "text-decoration-style"

### DISCUSSION THREAD

**Discussion by yannicka** [tailwindlabs/tailwindcss#3749] [Ideas]
It would be nice to be able to specify the underline style: https://developer.mozilla.org/fr/docs/Web/CSS/text-decoration-style

Currently, we are limited to "underlined" (`.underline`), "crossed out" (`.line-through`) and "not underlined" (`.no-underline`).

**Comment by dance2die** [comment]
As a workaround, you can create a plugin to add support.

https://play.tailwindcss.com/Wjwo66GH3G?file=config

With the plugin above, you can apply the style like

```html
<p class="underline text-decoration-style-dashed">...</p>
```

If you find yourself using those styles often, you could create a package and share w/ folks :wink: 

**Source** (in case the playground is deleted)

```js
const plugin = require('tailwindcss/plugin')

function textDecorationStyle({ addUtilities, theme, e }) {
  const values = theme('textDecorationStyle')

  const utility = Object.keys(values).map((key) => {
    return {
      [`.${e(`text-decoration-style-${key}`)}`]: {
        'text-decoration-style': values[key],
      },
    }
  })

  addUtilities(utility)
}

const textDecorationStylePlugin = plugin(textDecorationStyle, {
  theme: {
    textDecorationStyle: {
      solid: 'solid',
      double: 'double',
      dotted: 'dotted',
      dashed: 'dashed',
      wavy: 'wavy',
    },
  },
})

module.exports = {
  theme: {},
  variants: {},
  plugins: [textDecorationStylePlugin],
}

```

**Reply by yannicka** [reply]
Good workaround, thanks :)

**Reply by dance2die** [reply]
yw and enjoy~

**Comment by dance2die** [comment]
**December 10, 2021**

`@downwindcss/text-decoration` is deprecated starting Tailwind CSS v3 because TW now supports this out of the box.

---

I created a small plugin, [@downwindcss/text-decoration](https://www.npmjs.com/package/@downwindcss/text-decoration) (alpha version) to implement the said plugin.

Implemented `text-decoration-style` for now, but plan to add `text-decoration-color` and `text-decoration-thickness` as well.

You can see the demo here. https://codesandbox.io/s/downwindcsstext-decoration-demo-psjgs

---  

Forgot to add the repo in package.json.  Here is the source: https://github.com/downwindcss/text-decoration
